client/src/components/Header/components/language-list.tsx

```diff
@@ -0,0 +1,166 @@
+  LangNames,
+  LangCodes,
+  hiddenLangs
+} from '../../../../../config/i18n';
+import { hardGoTo as navigate } from '../../../redux/actions';
+import createLanguageRedirect from '../../create-language-redirect';
+import LanguageGlobe from '../../../assets/icons/language-globe';
+
+const locales = availableLangs.client.filter(
+  lang => !hiddenLangs.includes(lang)
+);
+
+const mapDispatchToProps = {
+  navigate
+};
+
+interface LanguageListProps {
+  t: TFunction;
+  navigate?: (location: string) => void;
+}
+
+export const LanguageList = ({
+  t,
+  navigate
+}: LanguageListProps): JSX.Element => {
+  const [showList, setShowList] = useState(false);
+  const listButtonRef = useRef<HTMLButtonElement>(null);
+
+  const handleClick = (): void => {
+    if (showList) {
+      setShowList(false);
+      return;
+    }
+    setShowList(true);
+  };
+
+  const handleLanguageChange = (
+    event: React.MouseEvent<HTMLButtonElement>
+  ): void => {
+    const selectedLanguage = event.currentTarget.dataset.value;
+    const isSelecetedCurrentLanguage =
+      selectedLanguage === clientLocale || selectedLanguage === undefined;
+
+    event.preventDefault();
+    listButtonRef.current?.focus();
+    setShowList(false);
+    if (isSelecetedCurrentLanguage) {
+      return;
+    }
+    const path = createLanguageRedirect({
+      clientLocale,
+      lang: selectedLanguage
+    });
+    if (navigate) {
+      return navigate(path);
+    }
+  };
+
+  const handleMenuKeyDown = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    if (event.key === 'Escape') {
+      listButtonRef.current?.focus();
+      setShowList(false);
+      event.preventDefault();
+    }
+  };
+
+  const handleBlur = (event: React.FocusEvent<HTMLButtonElement>) => {
+    if (
+      event.relatedTarget &&
+      !event.relatedTarget.closest('.nav-list') &&
+      event.relatedTarget !== listButtonRef.current
+    ) {
+      setShowList(false);
+    }
+  };
+
+  const handleLastLangaugeKeys = (
+    event: React.KeyboardEvent<HTMLButtonElement | HTMLAnchorElement>
+  ) => {
+    const DoKeyPress = new Map<string, { select: () => void }>([
+      [
+        'Escape',
+        {
+          select: () => {
+            listButtonRef.current?.focus();
+            setShowList(false);
+            event.preventDefault();
+          }
+        }
+      ],
+      [
+        'Tab',
+        {
+          select: () => {
+            const camperPressedTheShiftKey = event.shiftKey;
+            if (!camperPressedTheShiftKey) {
+              setShowList(false);
+            }
+          }
+        }
+      ]
+    ]);
+    DoKeyPress.get(event.key)?.select();
+  };
+
+  const getHandleLanguageKeys = (languagePosition: number) => {
+    const lastLanguage = locales.length - 1;
+    if (languagePosition === lastLanguage) {
+      return handleLastLangaugeKeys;
+    } else handleMenuKeyDown;
+  };
+  return (
+    <>
+      <button
+        id='toggle-button-nav'
```

Comment:
We can't use `toggle-button-nav` for the `id` because the main Menu button already uses that. Whatever you change it to here make sure to update `aria-labelledby` on the `ul` below.